packages/viewer/src/types/frame.ts

```diff
@@ -48,12 +66,24 @@ export const fromProto = (
     },
     sceneAttributes: {
       camera: FrameCamera.create({
-        position: Vector3.create(sceneAttributes.camera.position),
-        lookAt: Vector3.create(sceneAttributes.camera.lookAt),
-        up: Vector3.create(sceneAttributes.camera.up),
+        position: Vector3.create({
+          x: sceneAttributes.camera.position?.x || undefined,
+          y: sceneAttributes.camera.position?.y || undefined,
+          z: sceneAttributes.camera.position?.z || undefined,
+        }),
+        lookAt: Vector3.create({
+          x: sceneAttributes.camera.lookAt?.x || undefined,
+          y: sceneAttributes.camera.lookAt?.y || undefined,
+          z: sceneAttributes.camera.lookAt?.z || undefined,
+        }),
+        up: Vector3.create({
+          x: sceneAttributes.camera.up?.x || undefined,
+          y: sceneAttributes.camera.up?.y || undefined,
+          z: sceneAttributes.camera.up?.z || undefined,
+        }),
```

Comment:
I'm not too great at typescript types, but sensing this being necessary is a sign a type could be more useful.

Reply:
These are actually both typed. The issue here, I believe is related to an above comment on strict null types. This may be one argument for removing that strict rule. This code is trying to set the type `(number|null);` to `(number | undefined)` and it is unable to do so. Removing the strict rule would prevent these changes from being necessary.

Reply:
@jdm717 @danschultz What are your guys thoughts on this rule?

Reply:
Personally I think I'd prefer to leave that rule enabled. It adds a bit of overhead, but it seems like it's worth the tradeoff to avoid some pretty easy to introduce app-breaking bugs

Reply:
Because of how type coercion works in JS, if `camera.up.z = 0`, this will cause the returned value to be undefined. You should use `??` instead of `||` here.